app/scripts/directives/istagSelect.js

```diff
@@ -19,123 +20,133 @@ angular.module("openshiftConsole")
-            if (!_.find( $scope.isByNamespace[ns][$scope.istag.imageStream].status.tags, {tag: $scope.istag.tagObject.tag})) {
-              $scope.isByNamespace[ns][$scope.istag.imageStream].status.tags.push({tag: $scope.istag.tagObject.tag});
+      if (!_.includes(ctrl.namespaces, ns)) {
+        ctrl.namespaces.push(ns);
+        ctrl.isNamesByNamespace[ns] = ctrl.isNamesByNamespace[ns].concat(ctrl.istag.imageStream);
+        ctrl.isByNamespace[ns][ctrl.istag.imageStream] = {status: {tags: [{tag: ctrl.istag.tagObject.tag}]}};
+        return;
+      }
+      DataService.list(imageStreamsVersion, { namespace: ns }, function(isData) {
+        // Make a copy since we modify status tags and don't want to mutate objects that are cached.
+        var imageStreams = angular.copy(isData.by('metadata.name'));
+        ensureStatusTags(imageStreams);
+        ctrl.isByNamespace[ns] = imageStreams;
+        ctrl.isNamesByNamespace[ns] = _.keys(imageStreams).sort();
+
+        //  Image stream is missing
+        if (!_.includes(ctrl.isNamesByNamespace[ns], ctrl.istag.imageStream)) {
+          ctrl.isNamesByNamespace[ns] = ctrl.isNamesByNamespace[ns].concat(ctrl.istag.imageStream);
+          ctrl.isByNamespace[ns][ctrl.istag.imageStream] = {
+            status: {
+              tags: {}
             }
-          });
-        };
-
-        ProjectsService.list().then(function(projectData) {
-          $scope.namespaces = _.keys(projectData.by('metadata.name'));
+          };
+        }
+
+        // Tag is missing
+        if (!_.find( ctrl.isByNamespace[ns][ctrl.istag.imageStream].status.tags, {tag: ctrl.istag.tagObject.tag})) {
+          ctrl.isByNamespace[ns][ctrl.istag.imageStream].status.tags.push({tag: ctrl.istag.tagObject.tag});
+        }
+      });
+    };
 
-          if ($scope.includeSharedNamespace) {
-            // Use _.uniq to avoid adding "openshift" twice if the user is a
-            // member of the openshift namespace.
-            $scope.namespaces = _.uniq(['openshift'].concat($scope.namespaces));
-          }
+    ProjectsService.list().then(function(projectData) {
+      ctrl.namespaces = _.keys(projectData.by('metadata.name'));
 
-          $scope.namespaces = $scope.namespaces.sort();
+      if (ctrl.includeSharedNamespace) {
+        // Use _.uniq to avoid adding "openshift" twice if the user is a
+        // member of the openshift namespace.
+        ctrl.namespaces = _.uniq(['openshift'].concat(ctrl.namespaces));
+      }
 
-          // Fetch image streams when a new namespace is picked.
-          $scope.$watch('istag.namespace', function(namespace) {
-            if (!namespace || $scope.isByNamespace[namespace]) {
-              // We already have the data (or nothing selected).
-              return;
-            }
+      ctrl.namespaces = ctrl.namespaces.sort();
+
+      // Fetch image streams when a new namespace is picked.
+      $scope.$watch('$ctrl.istag.namespace', function(namespace) {
```

Comment:
We should be using `ng-change` for this instead of `$scope.$watch`. Ideally we remove all references to `$scope` if possible since it doesn't exist in Angular2

Reply:
was a bit surprised that `ng-change` was just setting istag fields to `null`. Will rework the logic in this PR.